numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -2434,39 +2467,51 @@ def launch_kernel(
     args,
     cooperative=False,
 ):
-    param_ptrs = [addressof(arg) for arg in args]
-    params = (c_void_p * len(param_ptrs))(*param_ptrs)
-
-    params_for_launch = addressof(params)
-    extra = 0
-
-    if cooperative:
-        driver.cuLaunchCooperativeKernel(
-            cufunc_handle,
-            gx,
-            gy,
-            gz,
-            bx,
-            by,
-            bz,
-            sharedmem,
-            hstream,
-            params_for_launch,
-        )
+    # Convert stream handle to cuda.core Stream object
+    if hstream is None:
+        hstream = 0
+    elif isinstance(hstream, binding.CUstream):
+        hstream = get_cuda_native_handle(hstream)
+    elif not isinstance(hstream, int):
+        hstream = 0
+    stream = ExperimentalStream.from_handle(hstream)
+
+    # Configure kernel launch parameters
+    config = LaunchConfig(
+        grid=(int(gx), int(gy), int(gz)),
+        block=(int(bx), int(by), int(bz)),
+        shmem_size=int(sharedmem) if sharedmem is not None else 0,
+        cooperative_launch=cooperative,
+    )
+
+    # Convert function handle to CUfunction binding type
+    if isinstance(cufunc_handle, binding.CUfunction):
+        cufunction = cufunc_handle
     else:
-        driver.cuLaunchKernel(
-            cufunc_handle,
-            gx,
-            gy,
-            gz,
-            bx,
-            by,
-            bz,
-            sharedmem,
-            hstream,
-            params_for_launch,
-            extra,
-        )
+        cufunction = binding.CUfunction(int(cufunc_handle))
+
+    # Convert arguments: cuda.core.launch expects numpy.uint64 for pointers,
+    # but can handle ctypes scalars directly
+    converted_args = [
+        np.uint64(arg.value if arg.value is not None else 0)
```

Comment:
```suggestion
        (arg.value if arg.value is not None else 0)
```
it'd be nice for the launch dispatcher in cuda.core to support c_void_p.